presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -601,6 +601,27 @@ public void testPushdownSubfields()
         assertPushdownSubfields("SELECT x.a, x.b, x.A + 2 FROM test_pushdown_struct_subfields WHERE x.B LIKE 'abc%'", "test_pushdown_struct_subfields",
                 ImmutableMap.of("x", toSubfields("x.a", "x.b")));
 
+        //Subfield pruning should pass-through arbitrary() function
```

Comment:
- would you move the new aggregation tests into `// Aggregation` section that starts on L566?
- this query is quite complex and it is difficult to see what's exactly being tested; consider breaking it up into simpler queries that test one "use case" at a time

```
- arbitrary(x).a
- arbitrary(x.a)
- arbitrary(x).d.d1
- arbitrary(x.d).d1
- ...
```

- let's also add similar tests for arrays and maps to testPushdownArraySubscripts and testPushdownMapSubscripts respectively (this may remove the need to change the schema of the test_pushdown_struct_subfields table)

Reply:
I have split the tests as you suggested. However, I think the combination exercises code paths which would not be exercised otherwise. Hence, there is utility in keeping these testcases. In this case, I discovered a bug in the earlier implementation where the entries were stored in a map and a column with different subfields had the old entries overwritten. I moved these tests to `testPushdownSubfieldsAssorted`. Please let me know if you feel otherwise and I can remove them.